tests/plugins/test_orchestrate_build.py

```diff
@@ -1138,48 +1182,144 @@ def test_orchestrate_override_build_kwarg(tmpdir, overrides):
+     "status": {"config": {"kind": "BuildConfig", "name": "wrong build config"}}}',
+     "Build config not found :",
+     False, None, None, None),
+
+    ('{"spec": {"strategy": {"customStrategy": {"from": \
+     {"name": "osbs-buildroot:latest", "kind": "DockerImage"}}}}, \
+     "status": {"config": {"kind": "BuildConfig", "name": "build config"}}}',
+     "BuildConfig object is malformed",
+     True, {"spec": {"strategy": {"customStrategy": {}}}}, None, None),
+
+    ('{"spec": {"strategy": {"customStrategy": {"from": \
+     {"name": "osbs-buildroot:latest", "kind": "DockerImage"}}}}, \
+     "status": {"config": {"kind": "BuildConfig", "name": "build config"}}}',
+     "BuildConfig object has unknown 'kind'",
+     True, {"spec": {"strategy": {"customStrategy": {"from": {"kind": "wrong_kind"}}}}},
+     None, None),
+
+    ('{"spec": {"strategy": {"customStrategy": {"from": \
+     {"name": "osbs-buildroot:latest", "kind": "DockerImage"}}}}, \
+     "status": {"config": {"kind": "BuildConfig", "name": "build config"}}}',
+     "ImageStreamTag not found",
+     True, {"spec": {"strategy": {"customStrategy": {"from": {"kind": "ImageStreamTag",
+                                                              "name": "wrong_ims"}}}}},
+     False, None),
+
+    ('{"spec": {"strategy": {"customStrategy": {"from": \
+     {"name": "osbs-buildroot:latest", "kind": "DockerImage"}}}}, \
+     "status": {"config": {"kind": "BuildConfig", "name": "build config"}}}',
+     "ImageStreamTag is malformed",
+     True, {"spec": {"strategy": {"customStrategy": {"from": {"kind": "ImageStreamTag",
+                                                              "name": "ims"}}}}},
+     True, {"image": {}}),
+
+    ('{"spec": {"strategy": {"customStrategy": {"from": \
+     {"name": "osbs-buildroot:latest", "kind": "DockerImage"}}}}, \
+     "status": {"config": {"kind": "BuildConfig", "name": "build config"}}}',
+     "Image in imageStreamTag 'ims' is missing Labels",
+     True, {"spec": {"strategy": {"customStrategy": {"from": {"kind": "ImageStreamTag",
+                                                              "name": "ims"}}}}},
+     True, {"image": {"dockerImageReference": "some@sha256:12345",
+                      "dockerImageMetadata": {"Config": {}}}}),
+
+    ('{"spec": {"strategy": {"customStrategy": {"from": \
+     {"name": "osbs-buildroot:latest", "kind": "DockerImage"}}}}, \
+     "status": {"config": {"kind": "BuildConfig", "name": "build config"}}}',
+     "Buildroot image doesn't have tag, which is needed for specified arch",
+     True, {"spec": {"strategy": {"customStrategy": {"from": {"kind": "ImageStreamTag",
+                                                              "name": "ims"}}}}},
+     True, {"image": {"dockerImageReference": "some@sha256:12345",
+                      "dockerImageMetadata": {
+                          "Config": {"Labels": {"release": "@sha256:123",
```

Comment:
The 'release' label won't look like that though, more like '4'.

Reply:
yes I know, but I need scenario where I have @256 and this was way how to get it

Reply:
I'll change it, to get it from annotations